### DISCUSSION TITLE
podman run failed - container create failed

### DISCUSSION THREAD

**Discussion by x2byue** [containers/podman#22567] [Q&A]
Podman version 4.6.1 on RHEL 8.9
concom version 2.18

podman run --rm -it test
Error: container create failed (no logs from conmon): conmon bytes "": readObjectStart: expect { or n, but found , error found in #0 byte of ...||..., bigger context ...||...

podman info:
host:
  arch: amd64
  buildahVersion: 1.31.5
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.8-1.module+el8.9.0+21697+6a5e98e7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.8, commit: 57ed23ee47beaf9a72b28f5666fab095a9ad4a38'
  cpuUtilization:
    idlePercent: 99.23
    systemPercent: 0.34
    userPercent: 0.42
  cpus: 8
  databaseBackend: boltdb
  distribution:
    distribution: '"rhel"'
    version: "8.9"
  eventLogger: file
  freeLocks: 2048
  hostname: crcrhel.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 60000
      size: 1
    - container_id: 1
      host_id: 60001
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 60000
      size: 1
    - container_id: 1
      host_id: 60001
      size: 65536
  kernel: 4.18.0-513.24.1.el8_9.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 30740901888
  memTotal: 33392648192
  networkBackend: cni
  networkBackendInfo:
    backend: cni
    dns:
      package: podman-plugins-4.6.1-9.module+el8.9.0+21697+6a5e98e7.x86_64
      path: /usr/libexec/cni/dnsname
      version: |-
        CNI dnsname plugin
        version: 1.3.1
        commit: unknown
    package: containernetworking-plugins-1.3.0-8.module+el8.9.0+21697+6a5e98e7.x86_64
    path: /usr/libexec/cni
  ociRuntime:
    name: runc
    package: runc-1.1.12-1.module+el8.9.0+21697+6a5e98e7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.12
      spec: 1.0.2-dev
      go: go1.20.12
      libseccomp: 2.5.2
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    path: /run/user/60000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.1-1.module+el8.9.0+21697+6a5e98e7.x86_64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 16905138176
  swapTotal: 16905138176
  uptime: 1h 36m 47.00s (Approximately 0.04 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/dmadminua/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/dmadminua/.local/share/containers/storage
  graphRootAllocated: 251092029440
  graphRootUsed: 53097222144
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 11
  runRoot: /run/user/60000/containers
  transientStore: false
  volumePath: /home/dmadminua/.local/share/containers/storage/volumes
version:
  APIVersion: 4.6.1
  Built: 1713524958
  BuiltTime: Fri Apr 19 04:09:18 2024
  GitCommit: ""
  GoVersion: go1.20.12
  Os: linux
  OsArch: linux/amd64
  Version: 4.6.1

podman run --log-level debug --rm -it test

INFO[0000] podman filtering at log level debug          
DEBU[0000] Called run.PersistentPreRunE(podman run --log-level debug --rm -it test) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/dmadminua/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/dmadminua/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/60000/containers    
DEBU[0000] Using static dir /home/dmadminua/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/60000/libpod/tmp     
DEBU[0000] Using volume path /home/dmadminua/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=xfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 25             
DEBU[0000] Successfully loaded 1 networks               
DEBU[0000] Pulling image test (policy: missing)         
DEBU[0000] Looking up image "test" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/000-shortnames.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/001-rhel-shortnames.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/002-rhel-shortnames-overrides.conf" 
DEBU[0000] Trying "localhost/test:latest" ...           
DEBU[0000] parsed reference into "[overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Found image "test" as "localhost/test:latest" in local containers storage 
DEBU[0000] Found image "test" as "localhost/test:latest" in local containers storage ([overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa) 
DEBU[0000] exporting opaque data as blob "sha256:cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Looking up image "localhost/test:latest" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "localhost/test:latest" ...           
DEBU[0000] parsed reference into "[overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Found image "localhost/test:latest" as "localhost/test:latest" in local containers storage 
DEBU[0000] Found image "localhost/test:latest" as "localhost/test:latest" in local containers storage ([overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa) 
DEBU[0000] exporting opaque data as blob "sha256:cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Looking up image "test" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "localhost/test:latest" ...           
DEBU[0000] parsed reference into "[overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Found image "test" as "localhost/test:latest" in local containers storage 
DEBU[0000] Found image "test" as "localhost/test:latest" in local containers storage ([overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa) 
DEBU[0000] exporting opaque data as blob "sha256:cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Inspecting image cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa 
DEBU[0000] exporting opaque data as blob "sha256:cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Inspecting image cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa 
DEBU[0000] Inspecting image cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa 
DEBU[0000] Inspecting image cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa 
DEBU[0000] Inspecting image cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa 
DEBU[0000] using systemd mode: true                     
DEBU[0000] No hostname set; container's hostname will default to runtime default 
DEBU[0000] Loading seccomp profile from "/usr/share/containers/seccomp.json" 
DEBU[0000] Allocated lock 0 for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] parsed reference into "[overlay@/home/dmadminua/.local/share/containers/storage+/run/user/60000/containers]@cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] exporting opaque data as blob "sha256:cf38a403a81b1654ae62c19fb34160518dae3c63323e05972f4371250b489daa" 
DEBU[0000] Cached value indicated that idmapped mounts for overlay are not supported 
DEBU[0000] Check for idmapped mounts support            
DEBU[0000] Created container "2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea" 
DEBU[0000] Container "2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea" has work directory "/home/dmadminua/.local/share/containers/storage/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata" 
DEBU[0000] Container "2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea" has run directory "/run/user/60000/containers/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata" 
DEBU[0000] Handling terminal attach                     
INFO[0000] Received shutdown.Stop(), terminating!        PID=6080
DEBU[0000] Enabling signal proxying                     
DEBU[0000] Cached value indicated that volatile is being used 
DEBU[0000] overlay: mount_data=lowerdir=/home/dmadminua/.local/share/containers/storage/overlay/l/2SGRA3U6THX2BPDLDPNP3BXYHP:/home/dmadminua/.local/share/containers/storage/overlay/l/BNZGDJE3DRWIOXHUIDCB2YRPBY:/home/dmadminua/.local/share/containers/storage/overlay/l/NKAMTO2XNQQIYZCXBTD4HZZPFN:/home/dmadminua/.local/share/containers/storage/overlay/l/L6XSRN4PLNIKMRSNAYWJGJJIDW:/home/dmadminua/.local/share/containers/storage/overlay/l/CP5RH5QEPQ5ZAPYI76OXRVPYSK:/home/dmadminua/.local/share/containers/storage/overlay/l/JOAYRFCDKIIEF24OZFGW2UDGD7:/home/dmadminua/.local/share/containers/storage/overlay/l/HVJAKTZD222I3MMJ6EFJFFASAM:/home/dmadminua/.local/share/containers/storage/overlay/l/NFFJZB47454BZYTY6TQ3YI45NL:/home/dmadminua/.local/share/containers/storage/overlay/l/2KOIS5MF2YEOVKRMLSACDPV4E5,upperdir=/home/dmadminua/.local/share/containers/storage/overlay/a3593d5a87ef65f5060aedf1d47221dfedacbc4a7bec72c8a0e18bd8473ad474/diff,workdir=/home/dmadminua/.local/share/containers/storage/overlay/a3593d5a87ef65f5060aedf1d47221dfedacbc4a7bec72c8a0e18bd8473ad474/work,,userxattr,volatile 
DEBU[0000] Mounted container "2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea" at "/home/dmadminua/.local/share/containers/storage/overlay/a3593d5a87ef65f5060aedf1d47221dfedacbc4a7bec72c8a0e18bd8473ad474/merged" 
DEBU[0000] Created root filesystem for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea at /home/dmadminua/.local/share/containers/storage/overlay/a3593d5a87ef65f5060aedf1d47221dfedacbc4a7bec72c8a0e18bd8473ad474/merged 
DEBU[0000] Made network namespace at /run/user/60000/netns/netns-7ed8bdde-5a00-13ed-62b8-bc592ec43f58 for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] slirp4netns command: /usr/bin/slirp4netns --disable-host-loopback --mtu=65520 --enable-sandbox --enable-seccomp --enable-ipv6 -c -r 3 -e 4 --netns-type=path /run/user/60000/netns/netns-7ed8bdde-5a00-13ed-62b8-bc592ec43f58 tap0 
DEBU[0000] /etc/system-fips does not exist on host, not mounting FIPS mode subscription 
DEBU[0000] Setting Cgroups for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea to user.slice:libpod:2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] reading hooks from /usr/share/containers/oci/hooks.d 
DEBU[0000] Workdir "/" resolved to host path "/home/dmadminua/.local/share/containers/storage/overlay/a3593d5a87ef65f5060aedf1d47221dfedacbc4a7bec72c8a0e18bd8473ad474/merged" 
DEBU[0000] Created OCI spec for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea at /home/dmadminua/.local/share/containers/storage/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata/config.json 
DEBU[0000] /usr/bin/conmon messages will be logged to syslog 
DEBU[0000] running conmon: /usr/bin/conmon               args="[--api-version 1 -c 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea -u 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea -r /usr/bin/runc -b /home/dmadminua/.local/share/containers/storage/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata -p /run/user/60000/containers/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata/pidfile -n relaxed_bhabha --exit-dir /run/user/60000/libpod/tmp/exits --full-attach -s -l k8s-file:/home/dmadminua/.local/share/containers/storage/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata/ctr.log --log-level debug --syslog -t --conmon-pidfile /run/user/60000/containers/overlay-containers/2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /home/dmadminua/.local/share/containers/storage --exit-command-arg --runroot --exit-command-arg /run/user/60000/containers --exit-command-arg --log-level --exit-command-arg debug --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/user/60000/libpod/tmp --exit-command-arg --network-config-dir --exit-command-arg  --exit-command-arg --network-backend --exit-command-arg cni --exit-command-arg --volumepath --exit-command-arg /home/dmadminua/.local/share/containers/storage/volumes --exit-command-arg --db-backend --exit-command-arg boltdb --exit-command-arg --transient-store=false --exit-command-arg --runtime --exit-command-arg runc --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg --events-backend --exit-command-arg file --exit-command-arg --syslog --exit-command-arg container --exit-command-arg cleanup --exit-command-arg --rm --exit-command-arg 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea]"
INFO[0000] Running conmon under slice user.slice and unitName libpod-conmon-2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea.scope 
DEBU[0000] Cleaning up container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] Tearing down network namespace at /run/user/60000/netns/netns-7ed8bdde-5a00-13ed-62b8-bc592ec43f58 for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] Unmounted container "2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea" 
DEBU[0000] Removing container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] Cleaning up container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] Container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea storage is already unmounted, skipping... 
DEBU[0000] Removing all exec sessions for container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea 
DEBU[0000] Container 2b43ceaf9556ad40e75d21f6841e282b8b230512c0f4d8ada35bdb0cd1e807ea storage is already unmounted, skipping... 
DEBU[0000] ExitCode msg: "container create failed (no logs from conmon): conmon bytes \"\": readobjectstart: expect { or n, but found \x00, error found in #0 byte of ...||..., bigger context ...||..." 
Error: container create failed (no logs from conmon): conmon bytes "": readObjectStart: expect { or n, but found , error found in #0 byte of ...||..., bigger context ...||...
DEBU[0000] Shutting down engines

**Comment by secondspass** [comment]
Based on discussion in https://github.com/containers/podman/issues/21024, this issue is fixed in conmon 2.1.10
